exec/bigmachine.go

```diff
@@ -1091,7 +1106,13 @@ func (w *worker) CommitCombiner(ctx context.Context, key TaskName, _ *struct{})
 		case combinerCommitted:
 			return nil
 		case combinerError:
-			return errors.E("error while writing combiner")
+			err := w.combinerErrors[key]
+			if strings.HasPrefix(err.Error(), "gob:") {
```

Comment:
👍 to @mariusae 's comment about moving this interpretation/translation closer to the call to `Encode`.  Then the error can just be marked with `maybeTaskFatalErr` here.

Reply:
(Also, we need to be careful not to wrap it with `fmt.Errorf` lest we lose the `*errors.Error` fatal-ness).